pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -668,6 +668,7 @@ private int getOrCreateDocId(Map<String, Object> dictIdMap) {
     int i = 0;
     int[] dictIds = new int[_numKeyColumns]; // dimensions + time column.
 
+    // TODO: this for loop breaks for multi value columns. https://github.com/apache/incubator-pinot/issues/3867
```

Comment:
we dont need this TODO anymore right?

Reply:
No, this TODO is to make the actual support for MV dimensions in RT aggregation of metrics. This PR is just disabling it in presence of MV columns.

Reply:
well, as the code stands it will never break since we disable if we have MV dimensions...but that is my ocd i guess :)

Reply:
i changed TODO to FIXME